### PR TITLE
chore(docs): Add "Gentics Mesh" guide

### PR DESCRIPTION
**Pull request by Jotschi**
## Description

This PR adds the sourcing guide for the OSS headless CMS [Gentics Mesh](https://getmesh.io)

### CODE REVIEW

**Comment by Jotschi** [comment]
@KyleAMathews I updated the PR. Now the checks are passing.

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+---
+title: Sourcing from Gentics Mesh
+---
+
+## Using Gentics Mesh
+
+[Gentics Mesh](https://getmesh.io) is an open-source self-hosted headless CMS that supports GraphQL. Contents like posts, products, images and more are represented by elements called "nodes". These nodes can then be queried with the familiar GraphQL syntax.
+
+The model of nodes can be defined within "schemas". In our demo we have schemas for vehicles, images and categories.
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]: Please try not to use "we" (https://www.gatsbyjs.org/contributing/gatsby-style-guide/#use-you-as-the-pronoun).

You could rephrase the sentence to make it more clear that with "we" Gentics Mesh is meant and that Gentics Mesh provides this demo etc.

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+## Getting started
+
+In this guide you'll create a complete project capable of querying data from a Gentics Mesh server.
+
+### Start Mesh
+
+Gentics Mesh is a self-hosted CMS. You can run it for example via docker:
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
Gentics Mesh is a self-hosted CMS. You can run it for example via [Docker](https://www.docker.com/):
```

Let's link to Docker in case people are not familiar with it

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+```shell
+docker run -p 8080:8080 -d gentics/mesh-demo
+```
+
+The UI can be accessed on http://localhost:8080/mesh-ui/ with default login (admin/admin).
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
The UI can be accessed on `http://localhost:8080/mesh-ui/` with default login (admin/admin).
```

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+### Install the boilerplate
+
+Now let's create a simple Gatsby starter site.
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
Now create a simple Gatsby starter site.
```

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+```shell
+gatsby new gatsby-site https://github.com/gatsbyjs/gatsby-starter-default
+```
+
+Navigate inside of the project with `cd gatsby-site`.
+
+### Add the Source Plugin
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
### Add the source plugin
```

Sentence case for headings :)

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+Additionally, you need the the `gatsby-source-graphql` library. Because Gentics Mesh uses GraphQL, you will take advantage of Gatsby's ability to simply stitch two GraphQL APIs together, reducing the time required to transform content. There is no need to use a special gatsby-source-x-cms plugin, the GraphQL source plugin is all you need.
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
Additionally, you need the `gatsby-source-graphql` plugin. Because Gentics Mesh uses GraphQL, you will take advantage of Gatsby's ability to stitch two GraphQL APIs together, reducing the time required to transform content. There is no need to use a special gatsby-source-x-cms plugin, the GraphQL source plugin is all you need.
```

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+You can install this component with:
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
Install the plugin:
```

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+```shell
+  npm install --save gatsby-source-graphql
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
npm install --save gatsby-source-graphql
```

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+```
+
+### Configure the plugin
+
+The last step required before you can query your data is to configure the `gatsby-source-graphql` plugin. Open `gatsby-config.js` and add the following object to the plugins array.
+This example uses the local Gentics Mesh Demo server that you have just started.
+
+```js
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
```javascript:title=gatsby-config.js
```

You can add titles to code blocks :)

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+From the root of your project, run the development environment with `npm run develop`. Once the server has started, you should be able to open the following URL in your browser:
+
+`http://localhost:8000/___graphql`
+
+This will show you an interface where you can test your new content GraphQL API.
+
+Try running this query:
+
+```js
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
```graphql
```

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+    }
+  }
+}
+```
+
+[You can read more about how to use the GraphQL API in the docs.](https://getmesh.io/docs/graphql/)
+
+### Getting content on the page
+
+For the purpose of this guide I've removed all the layout, SEO, link or other components in the Gatsby starter. The components are still there and 99% of users will likely want to put them back in once they understand what's happening in the code. Open the index file located at `src/pages/index.js` and replace the content with this code:
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]: Sorry, I don't follow but how is this paragraph relevant to the following parts? You're not mentioning or referencing the parts.

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+
+```js
```
Comment (LekoArts) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
```javascript:title=src/pages/index.js
```

File: docs/docs/sourcing-from-gentics-mesh.md (hunk 11 above)
Comment (Jotschi) [reply docs/docs/sourcing-from-gentics-mesh.md]: I simplified the wording and referenced the file that contains the mentioned components.

File: docs/docs/sourcing-from-gentics-mesh.md (hunk 1 above)
Comment (Jotschi) [reply docs/docs/sourcing-from-gentics-mesh.md]: Good point. I have rephrased the sentence.

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+        url: "http://localhost:8080/api/v2/demo/graphql",
+    },
+},
+```
+
+If everything works correctly, you should now have your Gentics Mesh API added to the Gatsby source API!
+
+### Querying for content
+
+From the root of your project, run the development environment with `npm run develop`. Once the server has started, you should be able to open the following URL in your browser:
```
Comment (muescha) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
From the root of your project, run the development environment with `gatsby develop`. Once the server has started, you should be able to open the following URL in your browser:
```

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+)
+
+export default IndexPage
+```
+
+Once saved you can access the page via `http://localhost:8000`
+
+You now have:
+
+1. Added the `StaticQuery` component to a page that allows you to fetch content from the GraphQL API.
```
Comment (muescha) [inline docs/docs/sourcing-from-gentics-mesh.md]: whats more suggested? the `useStaticQuery` or `StaticQuery`? i personally would prefer the hook

File: docs/docs/sourcing-from-graphcms.md [hunk 15]
```diff
@@ -56,7 +56,7 @@ If everything works correctly, you should now have your GraphCMS data added to t
 
 ### Querying for content
 
-From the root of your project, run the development environment with `npm run development`. Once the server has started and is error free, you should be able to open the following URL in your browser:
+From the root of your project, run the development environment with `npm run develop`. Once the server has started and is error free, you should be able to open the following URL in your browser:
```
Comment (muescha) [inline docs/docs/sourcing-from-graphcms.md]:
```suggestion
From the root of your project, run the development environment with `gatsby develop`. Once the server has started and is error free, you should be able to open the following URL in your browser:
```

**Comment by muescha** [comment]
i would also do some link titles a little shorter...

**Comment by muescha** [comment]
the guide should be linked from some places:
- sidebar
- https://www.gatsbyjs.org/docs/headless-cms/

**Comment by Jotschi** [comment]
> the guide should be linked from some places:
> 
>     * sidebar
> 
>     * https://www.gatsbyjs.org/docs/headless-cms/

I added an entry to the headless cms table but I don't know where the guide should be linked in the sidebar.

**Comment by LekoArts** [comment]
> the guide should be linked from some places:
> sidebar

@muescha That's not 100% accurate. We have a selected list of CMSs we deem popular/relevant and we want to highlight in the sidebar. Hence the inline comment:

https://github.com/gatsbyjs/gatsby/blob/master/www/src/data/sidebars/doc-links.yaml#L146-L148

Contributors can add the CMS + Guide in the list.

File: docs/docs/sourcing-from-gentics-mesh.md (hunk 14 above)
Comment (Jotschi) [reply docs/docs/sourcing-from-gentics-mesh.md]: I also prefer usage of hooks. The examples in the guides on static-query do however only show usage of `StaticQuery`. It might be better to keep docs consistent?

https://www.gatsbyjs.org/docs/static-query/

**Comment by muescha** [comment]
👍 

That's why I did not found all in the Sidebar. Thx for the info

**Comment by Jotschi** [comment]
@muescha  @LekoArts Are there any other open points that you want me to address?

File: docs/docs/sourcing-from-gentics-mesh.md [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+
+### Start Mesh
+
+Gentics Mesh is a self-hosted CMS. You can run it for example via [Docker](https://www.docker.com/):
+
+```shell
+docker run -p 8080:8080 -d gentics/mesh-demo
+```
+
+The UI can be accessed on `http://localhost:8080/mesh-ui/` with default login (admin/admin).
```
Comment (gillkyle) [inline docs/docs/sourcing-from-gentics-mesh.md]:
```suggestion
The demo Gentics server that gets setup has a UI that can be accessed on `http://localhost:8080/mesh-ui/` with default login (username: admin, password: admin). It allows you to view the content stored on the CMS and add new content.
```

**Comment by Jotschi** [comment]
@gillkyle Thanks. I have merged your suggestions.
